modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/StorageHashIndexDescriptor.java

```diff
@@ -112,17 +114,22 @@ public List<StorageHashIndexColumnDescriptor> columns() {
         return columns;
     }
 
+    @Override
+    public boolean isPk() {
```

Comment:
Why does it have to be an internal flag instead of the external comparison, like you do in the constructor? Seems excessive to me

Reply:
Please give an example of how you could implement your idea in those places where I use this internal flag.

Reply:
What do you mean? Many of those places I wouldn't even touch, code would be the same. My proposal - set nextRowIdToBuild to null when we create / recreate index that is PK. Explicitly and in the outside code

Reply:
What problems do I see in your campaign:
1) We need to remember that for the primary index we need to reset field `nextRowIdToBuild`, which may be forgotten after some time.
2) Performing these two operations is not atomic, it can lead to unpleasant consequences in the product code for users; the first thing that comes to mind is that the server will be pulled out of the socket immediately after the creation of the index storage is completed. Then we will need to add the correct recovery procedure somewhere else.

I don’t claim that my solution is ideal, I don’t really like it myself, but it looks more correct at the moment, I suggest leaving it.

Reply:
Let's leave it, with hopes that we will change it in the future.
> Performing these two operations is not atomic

We can make them atomic, what's the issue?

> which may be forgotten after some time.

same applies to your code

Reply:
> Let's leave it, with hopes that we will change it in the future.

Nice.

> We can make them atomic, what's the issue?

Yes, in general, we can, but so far in my head it looks better, plus writing tests in different places is not centralized.

> same applies to your code

Maybe.